src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -144,4 +155,115 @@ public String getAddForeignKeyConstraintString(
+    return false;
+  }
+
+  @Override
+  public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode) {
+    return LOCKING_STRATEGY;
+  }
+
+  @Override
+  public String getForUpdateString(LockOptions lockOptions) {
+    return "";
+  }
+
+  @Override
+  public String getForUpdateString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateString(String aliases, LockOptions lockOptions) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getWriteLockString(int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getWriteLockString(String aliases, int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getReadLockString(int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getReadLockString(String aliases, int timeout) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public boolean supportsOuterJoinForUpdate() {
+    return false;
+  }
+
+  @Override
+  public String getForUpdateNowaitString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateNowaitString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+
+  @Override
+  public String getForUpdateSkipLockedString() {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String getForUpdateSkipLockedString(String aliases) {
+    throw new UnsupportedOperationException("Cloud Spanner does not support selecting for lock"
+        + " acquisition.");
+  }
+
+  @Override
+  public String applyLocksToSql(String sql, LockOptions aliasedLockOptions,
+      Map<String, String[]> keyColumnNames) {
+    return sql;
+  }
+
+  /**
+   * A locking strategy for the Cloud Spanner dialect that does nothing. Cloud Spanner does not
+   * support locking.
+   *
+   * @author Chengyuan Zhao
+   */
+  public static class DoNothingLockingStrategy implements LockingStrategy {
```

Comment:
Remove the `public` modifier?

Reply:
done. I made it package private (so the test can see it)